### PR TITLE
Fix on xcode 6.4

### PR DESCRIPTION
**Pull request by LaserPatrick**
fix DVTPlugInCompatibilityUUIDs for xcode 6.4 (6E35b)
fix load menu after 3 sec else doesn't appear menu item

### CODE REVIEW

**Comment by LaserPatrick** [comment]
:) ok is not a very good workaround but on initWithBundle the [NSApp mainMenu] is nil. Ergo the plugin can add Tags button on Source Control menu item.

**Comment by wczekalski** [comment]
Is there any other way we can detect when -[NSApp mainMenu] is initialized? Like a notification? This call with a magic number is extremely unsafe.

**Comment by LaserPatrick** [comment]
ok i fixit now

File: WCGitTagsPlugin.xcodeproj/project.xcworkspace/xcshareddata/WCGitTagsPlugin.xccheckout [hunk 1]
```diff
@@ -7,21 +7,21 @@
...
-	<string>WCGitTagsPlugin.xcodeproj</string>
+	<string>WCGitTagsPlugin.xcodeproj/project.xcworkspace</string>
 	<key>IDESourceControlProjectRelativeInstallPathDictionary</key>
 	<dict>
 		<key>4A1D773003D37339838A9239E20BFBC1F14D7F74</key>
 		<string>../..</string>
 	</dict>
 	<key>IDESourceControlProjectURL</key>
-	<string>https://github.com/wczekalski/WCGitTagsPlugin.git</string>
+	<string>https://github.com/LaserPatrick/WCGitTagsPlugin.git</string>
```
Comment (wczekalski) [inline WCGitTagsPlugin.xcodeproj/project.xcworkspace/xcshareddata/WCGitTagsPlugin.xccheckout]: I don't think this should change...

File: WCGitTagsPlugin/WCGitTagsPlugin.m [hunk 2]
```diff
@@ -54,7 +54,7 @@ + (void)pluginDidLoad:(NSBundle *)plugin
     NSString *currentApplicationName = [[NSBundle mainBundle] infoDictionary][@"CFBundleName"];
     if ([currentApplicationName isEqual:@"Xcode"]) {
         dispatch_once(&onceToken, ^{
-            sharedPlugin = [[self alloc] initWithBundle:plugin];
+                sharedPlugin = [[self alloc] initWithBundle:plugin];
```
Comment (wczekalski) [inline WCGitTagsPlugin/WCGitTagsPlugin.m]: Wrong indentation

File: WCGitTagsPlugin/WCGitTagsPlugin.m [hunk 3]
```diff
@@ -65,39 +65,50 @@ - (id)initWithBundle:(NSBundle *)plugin
...
-            self.beingPresented = NO;
-        }
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(didApplicationFinishLaunchingNotification:)
+                                                     name:NSApplicationDidFinishLaunchingNotification
+                                                   object:nil];
+        
+
+        WCTagWatchdog *watchDog = [[WCTagWatchdog alloc] initWithWatchBlock:^{
+            [self willChangeValueForKey:@"tags"];
```
Comment (wczekalski) [inline WCGitTagsPlugin/WCGitTagsPlugin.m]: I know this is copy-paste of my code, but there's a memory leak here. `self` should be weak.

File: WCGitTagsPlugin/WCGitTagsPlugin.m [hunk 4]
```diff
@@ -65,39 +65,50 @@ - (id)initWithBundle:(NSBundle *)plugin
...
+            [[menuItem submenu] addItem:actionMenuItem];
+        } else {
+            [[menuItem submenu] insertItem:actionMenuItem atIndex:indexOfRefreshStatusItem];
+        }
+        
+        self.tagsItem = actionMenuItem;
+        [self.tagsItem setEnabled:self.refreshStatusItem.isEnabled];
+        
+    }
+}
```
Comment (wczekalski) [inline WCGitTagsPlugin/WCGitTagsPlugin.m]: Great!

File: WCGitTagsPlugin/WCGitTagsPlugin.m [hunk 5]
```diff
@@ -114,15 +125,16 @@ - (void)dealloc
...
+    if ([self.bundle loadNibNamed:@"View" owner:self topLevelObjects:&topLevelObjects]) {
+        for (id object in topLevelObjects) {
+            // Defensive way of doing things.
+            if ([object isKindOfClass:[NSWindow class]]) {
+                NSWindow *window = (NSWindow *)object;
+                if ([window.identifier isEqualToString:@"Add tag window"]) {
+                    self.addTagWindow = window;
+                } else if ([window.identifier isEqualToString:@"Tags window"]) {
+                    self.tagsWindow = window;
+                }
```
Comment (wczekalski) [inline WCGitTagsPlugin/WCGitTagsPlugin.m]: I don't understand these changes

File: WCGitTagsPlugin/WCGitTagsPlugin.m [hunk 6]
```diff
@@ -140,17 +152,23 @@ - (void)presentTagsModal:(id)sender {
...
+            
+            self.watchDog.gitDirectoryURL = self.repository.gitDirectoryURL;
+            [self.watchDog start];
+            [self syncTags];
+        } else {
+            NSAlert *alert = [[NSAlert alloc] init];
+            alert.alertStyle = NSWarningAlertStyle;
+            alert.messageText = @"Generic error.";
+            [alert beginSheetModalForWindow:[NSApp keyWindow] completionHandler:nil];
+        }
```
Comment (wczekalski) [inline WCGitTagsPlugin/WCGitTagsPlugin.m]: Don't understand these either. Could you explain them?

**Comment by wczekalski** [comment]
The rest is great. I will merge when the rest is discussed! 

Thanks a lot @LaserPatrick :+1:

File: WCGitTagsPlugin/WCGitTagsPlugin.m [hunk 7]
```diff
@@ -140,17 +151,18 @@ - (void)presentTagsModal:(id)sender {
...
-        [[NSApp keyWindow] beginSheet:self.tagsWindow completionHandler:^(NSModalResponse returnCode) {
-            [self.tagsWindow orderOut:self];
-            self.repository = nil;
-            self.beingPresented = NO;
-        }];
-        
-        self.watchDog.gitDirectoryURL = self.repository.gitDirectoryURL;
-        [self.watchDog start];
-        [self syncTags];
+        if (self.tagsWindow) {
```
Comment (LaserPatrick) [inline WCGitTagsPlugin/WCGitTagsPlugin.m]: i've added this check

if method loadInterface can't load correctly the nib file beingPresented Bool changed state

**Comment by wczekalski** [comment]
Ok, merging

**Pull request merged**